### DISCUSSION TITLE
Make Specific Tags Move File into Determined Folders

### DISCUSSION THREAD

**Discussion by hepheastus** [stashapp/stash#3332] [Q&A] [answered]
I have stash running in an unraid docker.  I have been running it for 3 days in total, I am new to this (4 years using freeNAS with general use).

I would like it if Stash can help with organization and I would love it if I can do it by tags.
Does stash have the ability to do this currently?

I tried searching with a certain tag, selecting all then using export to move tagged videos into a folder, all it seems to do is export tag information with JSON files.

Even better is if we can make specific tags, by editing them manually, to move files into determined folders.
Like main categories: CGI, JAV, General etc etc.

It also wouldn't be the worst thing to have to select clips, either manually or select all with a filtered search, and being able to select "move files".

**Comment by HereIam80** [comment, marked as answer]
There have been discussions before on this feature and it will most likely not be a part of stashs core functionality any time soon. The reason is that it is complicated to make it work on all OS. There is so much that can go wrong and many errors are likely to occur with permissions, etc.
Also stash doesn't really care about your folder structure. The whole point of stash is even to not to have to care about a folder structure at all.

That being said there is a plugin for stash which does exactly what you are looking for here: https://docs.stashapp.cc/docs/Add-ons/Plugins/ and it is called "renamerOnUpdate". At least I believe it does what you want as I don't use it personally.

**Reply by hepheastus** [reply]
I will attempt this plugin.
Thanks!  If I have any success I'll respond in this main thread, not as a reply.

**Comment by hepheastus** [comment]
As Herlam80 directed, 7 hours later I got renamerOnUpdate to work as a test on windows.  It wasn't exactly friendly to set up.  I never worked with Python or used commands to download modules, that took some effort.  The config file for the plugin also seems to assume previous knowledge.

For anybody who is as new as me, here is how I achieved it after python setup and "requests" module install -
I went to the "Move Your Files" portion in renameOnUpdate config file, turned the template to true, copied and pasted the $tags value into the p-default-template (removing performer value).  In whitelist, added the specific tags I only wanted to work and then ran scan, it worked in a manner that was suitable.

Ended up using it in a different manner than expected, I am using it to finely sort sub directories.  
I dumped everything into an Unsorted Folder, my main folders look something like this

Main Folder Sorts,
[Main Genre1] [Main Genre2] [Unsorted] [Main Genre3]
Genre being something like P, H, JAV or whatever.

For Sub Folder Sorts,
White listed things like Glasses, Stockings or whatever.

After tagging a few videos, hit scan, go into unsorted, use windows thumbnails or quickly previewing the video to figure out main folder sort.  I just threw those videos into the main folder sorts, hit scan again, they get resorted into sub folder sorts.
I had to do it this way because I don't know how to make the template in the config file do Main Folder Sort by tag then nest Sub Folder Sort by tag.  If I could do that, I could just cut and paste into the parent directory holding main folder sorts and kill a lot of steps.
